src/include/aws.hpp

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include "duckdb/common/string.hpp"
+#include "duckdb/main/client_context.hpp"
+
+namespace duckdb {
+
+class AWSInput {
+public:
+	AWSInput() {
+	}
+
+public:
+	string GetRequest(ClientContext &context);
+
+public:
+	//! NOTE: 'scheme' is assumed to be HTTPS!
+	string authority;
+	vector<string> path_segments;
+	vector<std::pair<string, string>> query_string_parameters;
+	string user_agent;
+	string cert_path;
+
+	//! Provider credentials
+	string key_id;
```

Comment:
I wonder if these should be stored here or not, or if there should be reference to the secret name instead. I could see a situation where some crazy error occurs during an AWS request and the stack gets dumped, which would expose these values. They are temporary, but maybe it's still smart to store these values somewhere else. 
Doesn't have to be in this PR though, I can see this being part of a larger rework